### PR TITLE
executor: add config EnableCollectExecutionInfo

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

### What problem does this PR solve?

This is the first PR of https://github.com/pingcap/tidb/issues/15008

### What is changed and how it works?

Add config `EnableCollectExecutionInfo`

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

Add config `EnableCollectExecutionInfo`

### CODE REVIEW

**Comment by TennyZhuang** [comment]
@sre-bot /bench

File: executor/builder.go [hunk 1]
```diff
@@ -95,6 +96,9 @@ type MockPhysicalPlan interface {
 }
 
 func (b *executorBuilder) build(p plannercore.Plan) Executor {
+	if config.GetGlobalConfig().EnableCollectExecutionInfo {
```
Comment (crazycs520) [inline executor/builder.go]: add `b.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl == nil` check to avoid repeat assignment

**Comment by TennyZhuang** [comment]
@sre-bot /run-all-tests

**Comment by TennyZhuang** [comment]
@sre-bot /run-all-tests

**Comment by qw4990** [comment]
Should we cherry-pick this PR to v4.0? @TennyZhuang @crazycs520

**Pull request merged**